### ISSUE TITLE
about accuracy

### ISSUE THREAD

**Issue by rehammadi** [XifengGuo/DEC-keras#11]
several run times lead to several accuracy values? what can be the problem of this issue?

**Comment by AlexandreLeboucher** [comment]
Same here, pretty confused about it as well

**Comment by jzuern** [comment]
Kmeans clustering has a per-default randomized centroid initialization procedure that affects accuracy (especially with ill-formed clusters) that is also used here. Try again with setting `KMeans(random_state=42)` and you should get deterministic clustering.

**Comment by XifengGuo** [comment]
Apart from kmeans mentioned by @jzuern , the autoencoder also gets to different states since the randomness exists in weight initialization and stochastic gradient descent optimization algorithm
